File: Modulo08/AlunoEndereco-Java/Aluno.java
public class Aluno {
    private String nome;
    private int RA;
    private Endereco endereco;

    public Aluno(String nome, int RA, Endereco endereco) {
        this.nome = nome;
        this.RA = RA;
        this.endereco = endereco;
    }

    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getRA() {
        return this.RA;
    }

    public void setRA(int RA) {
        this.RA = RA;
    }

    public Endereco getEndereco() {
        return this.endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }

    public void imprime() {
        System.out.println("-------------------------------------");
        System.out.println("Nome: " + this.getNome());
        System.out.println("RA: " + this.getRA());
        System.out.println("<Endereco>");
        this.getEndereco().imprime();
    }
}

File: Modulo08/AlunoEndereco-Referencia/Aluno.cpp
#include "Aluno.h"
#include <iostream>

using namespace std;

// construtor com 3 parâmetros

Aluno::Aluno(string nome, int RA, Endereco& endereco) : nome(nome), RA(RA),
endereco(endereco) {
}

// destrutor da classe

Aluno::~Aluno() {
    cout << "Destrutor Aluno (" << nome << ", " << RA << ")" << endl;
    
}

string Aluno::getNome() const {
    return nome;
}

void Aluno::setNome(string val) {
    nome = val;
}

int Aluno::getRA() const {
    return RA;
}

void Aluno::setRA(int val) {
    RA = val;
}

Endereco& Aluno::getEndereco() const {
    return endereco;
}

void Aluno::setEndereco(Endereco& val) {
    endereco = val;
}

void Aluno::imprime() const {
    cout << "-------------------------------------" << endl;
    cout << "Nome: " << this->getNome() << endl;
    cout << "RA: " << this->getRA() << endl;
    cout << "<Endereco>" << endl;
    this->getEndereco().imprime();
}